src/java/com/puppetlabs/puppetserver/MetricsPuppetProfiler.java

```diff
@@ -90,8 +176,8 @@ private boolean shouldTime(String[] metric_id) {
         return true;
     }
 
-    private List<Timer> getTimers(String[] metric_id) {
-        List<Timer> timers = new ArrayList<Timer>();
+    private Map<String, Timer> getOrCreateTimersByIDs(String[] metric_id) {
```

Comment:
This name is not used anywhere else right? Its not important for https://github.com/puppetlabs/puppetserver/blob/ebc41ebdb39690c597147e6a9bbbc66b8283bd97/src/clj/puppetlabs/services/jruby/jruby_metrics_core.clj#L212 ?

Reply:
I'm not sure i'm following the "or" in the name here. It seems like this just creates a new map and returns it unconditionally. Am I missing something?

Reply:
> This name is not used anywhere else right? Its not important for 

It's not, the method is private and external callers are calling this method on the MetricRegistry https://www.javadoc.io/doc/io.dropwizard.metrics/metrics-core/4.1.0-rc2/com/codahale/metrics/MetricRegistry.html#getTimers--

I think the two cases of "getTimers()" here where inspired by that method. But, the call to MetricRegister.timer() in this method is actually the place in this class were we _create_ timers (if they don't exist). See https://www.javadoc.io/doc/io.dropwizard.metrics/metrics-core/4.1.0-rc2/com/codahale/metrics/MetricRegistry.html#timer-java.lang.String-

The `getOrCreateTimers` is in reference to that. A more accurate name would probably be `getOrCreateTimersAndReturnThemById` , maybe?

Reply:
Ah, thanks for the reference. I think leaving it as is will be fine!